File: Clases 2018/Clases-C#/ClasePractica12_2018 (transacciones con data manager)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/EstadoDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class EstadoDao
    {
        public List<Estado> getAll()
        {
            List<Estado> lst = new List<Estado>();
            DataManager dm = new DataManager();
            try
            {
                dm.Open();
                foreach (DataRow row in dm.ConsultaSQL("Select * from Estados").Rows)
                    lst.Add(map(row));                
            }
            catch (Exception ex)
            {
                dm.Rollback();
            }
            finally
            {
                // Cierra la conexión 
                dm.Close();
            }
            return lst;
        
        }

        private Estado map(DataRow row)
        {
            Estado oEstado = new Estado();
            oEstado.id_estado = Convert.ToInt32(row["id_Estado"].ToString());
            oEstado.nombre = row["n_estado"].ToString();
            return oEstado;
        }
    }
}

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/EstadoDao.vb
Class EstadoDao
    Public Function getAll() As List(Of Estado)
        Dim lst As List(Of Estado) = New List(Of Estado)()
        Dim dm As DataManager = New DataManager()

        Try
            dm.Open()

            For Each row As DataRow In dm.ConsultaSQL("Select * from Estados").Rows
                lst.Add(map(row))
            Next

        Catch ex As Exception
            dm.Rollback()
        Finally
            dm.Close()
        End Try

        Return lst
    End Function

    Private Function map(ByVal row As DataRow) As Estado
        Dim oEstado As Estado = New Estado()
        oEstado.id_estado = Convert.ToInt32(row("id_Estado").ToString())
        oEstado.nombre = row("n_estado").ToString()
        Return oEstado
    End Function
End Class